ext/ffi/jit_trampoline.rs

```diff
@@ -82,16 +95,31 @@ pub(crate) fn codegen(sym: &crate::Symbol) -> String {
     c += native_arg_to_c(ty);
     let _ = write!(c, " p{i}");
   }
+  if needs_unwrap {
+    let _ = write!(c, ", struct FastApiTypedArray *p_ret");
```

Comment:
**nit:**

SysV ABI at least passes return value pointers as the 0th parameter. We can't quite get that far due to the receiver object (unless we piggy-back on that somehow) but I'd still maybe put the return value buffer as the first argument, instead of being the very last one. Just because.

But it doesn't really matter.

Reply:
Oh yeah, Windows ABI does the same for return value pointers now that I think about it.